lib/matplotlib/tests/test_image.py

```diff
@@ -151,16 +160,26 @@ def test_imsave_color_alpha():
     buff.seek(0)
     arr_buf = plt.imread(buff)
 
-    # Recreate the float -> uint8 -> float32 conversion of the data
-    data = (255*data).astype('uint8').astype('float32')/255
-    # Wherever alpha values were rounded down to 0, the rgb values all get set
-    # to 0 during imsave (this is reasonable behaviour).
-    # Recreate that here:
-    for j in range(3):
-        data[data[:, :, 3] == 0, j] = 1
+    # Recreate the float -> uint8 conversion of the data
+    # We can only expect to be the same with 8 bits of precision,
+    # since that's what the PNG file used.
+    data = (255*data).astype('uint8')
+    arr_buf = (255*arr_buf).astype('uint8')
 
     assert_array_equal(data, arr_buf)
 
```

Comment:
This is consistent with the rest of the file.  Better leave purely PEP8 updates to a separate PR.